packages/commons-test/src/setupTestContainersVitestGlobal.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable functional/immutable-data */
+/* eslint-disable functional/no-let */
+
+import { config as dotenv } from "dotenv-flow";
+import { StartedTestContainer } from "testcontainers";
+import type { provide } from "vitest";
+import {
+  TEST_MINIO_PORT,
+  TEST_MONGO_DB_PORT,
+  TEST_POSTGRES_DB_PORT,
+  TestContainersConfig,
+  minioContainer,
+  mongoDBContainer,
+  postgreSQLContainer,
+} from "./containerTestUtils.js";
+
+declare module "vitest" {
+  export interface ProvidedContext {
+    config: TestContainersConfig;
+  }
+  export type provide = <K extends keyof ProvidedContext>(
+    key: K,
+    value: ProvidedContext[K]
+  ) => void;
+}
+
+export function setupTestContainersVitestGlobal() {
+  dotenv();
+  const config = TestContainersConfig.parse(process.env);
+  let teardown = false;
```

Comment:
The teardown variable was due to a bug that led the cleanup function to be called twice. Investigating further I noticed that the bug has been fixed since vitest `0.30.0` ([related PR)](https://github.com/vitest-dev/vitest/issues/3181), so removing it now.